From 0b7591ff4635d5ec2476f5c2cc1611a6fc975669 Mon Sep 17 00:00:00 2001
From: Meir Levi <mlevi4@marvell.com>
Date: Thu, 24 Dec 2020 10:40:37 -0500
Subject: [PATCH 1069/1921] dts: marvell: musdk: add XOR UIO nodes

XOR used by user space networking applications.
Added 2 XOR UIO's for each CP and 4 UIO's for AP.

Signed-off-by: Meir Levi <mlevi4@marvell.com>
Change-Id: Ic210bf3f6d01b75148da42279823cd94a1de5eeb
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42777
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi | 28 ++++++++++++++++---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 10 +++++++
 2 files changed, 34 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index 7c0daadc2c66..82a79c6765c5 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -167,7 +167,7 @@
 				msi-controller;
 			};
 
-			xor@400000 {
+			xor0: xor@400000 {
 				compatible = "marvell,armada-7k-xor", "marvell,xor-v2";
 				reg = <0x400000 0x1000>,
 				      <0x410000 0x1000>;
@@ -177,7 +177,7 @@
 				iommus = <&smmu 0x1>;
 			};
 
-			xor@420000 {
+			xor1: xor@420000 {
 				compatible = "marvell,armada-7k-xor", "marvell,xor-v2";
 				reg = <0x420000 0x1000>,
 				      <0x430000 0x1000>;
@@ -187,7 +187,7 @@
 				iommus = <&smmu 0x2>;
 			};
 
-			xor@440000 {
+			xor2: xor@440000 {
 				compatible = "marvell,armada-7k-xor", "marvell,xor-v2";
 				reg = <0x440000 0x1000>,
 				      <0x450000 0x1000>;
@@ -196,7 +196,7 @@
 				dma-coherent;
 			};
 
-			xor@460000 {
+			xor3: xor@460000 {
 				compatible = "marvell,armada-7k-xor", "marvell,xor-v2";
 				reg = <0x460000 0x1000>,
 				      <0x470000 0x1000>;
@@ -205,6 +205,26 @@
 				dma-coherent;
 			};
 
+			uio_xor0 {
+				compatible = "marvell,uio-xor-v2";
+				xor_access = <&xor0>;
+			};
+
+			uio_xor1 {
+				compatible = "marvell,uio-xor-v2";
+				xor_access = <&xor1>;
+			};
+
+			uio_xor2 {
+				compatible = "marvell,uio-xor-v2";
+				xor_access = <&xor2>;
+			};
+
+			uio_xor3 {
+				compatible = "marvell,uio-xor-v2";
+				xor_access = <&xor3>;
+			};
+
 			spi0: spi@510600 {
 				compatible = "marvell,armada-380-spi";
 				reg = <0x510600 0x50>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index db1346ad1be0..aebe5820310e 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -340,6 +340,16 @@
 				 <&CP11X_LABEL(clk) 1 14>;
 		};
 
+		CP11X_LABEL(uio_xor0) {
+			compatible = "marvell,uio-xor-v2";
+			xor_access = <&CP11X_LABEL(xor0)>;
+		};
+
+		CP11X_LABEL(uio_xor1) {
+			compatible = "marvell,uio-xor-v2";
+			xor_access = <&CP11X_LABEL(xor1)>;
+		};
+
 		CP11X_LABEL(spi0): spi@700600 {
 			compatible = "marvell,armada-380-spi";
 			reg = <0x700600 0x50>;
-- 
2.31.1

